samples/dnn/dasiamrpn_tracker.py

```diff
@@ -0,0 +1,295 @@
+                te_im[r + top_pad:, left_pad:left_pad + c, :] = self.avg_chans
+            if left_pad:
+                te_im[:, 0:left_pad, :] = self.avg_chans
+            if right_pad:
+                te_im[:, c + left_pad:, :] = self.avg_chans
+            im_patch_original = te_im[int(context_ymin):int(context_ymax + 1), int(context_xmin):int(context_xmax + 1), :]
+        else:
+            im_patch_original = im[int(context_ymin):int(context_ymax + 1), int(context_xmin):int(context_xmax + 1), :]
+
+        if not np.array_equal(model_size, original_sz):
+            im_patch = cv.resize(im_patch_original, (model_size, model_size))
+        else:
+            im_patch = im_patch_original
+
+        return im_patch.astype(float)
+
+def main():
+    parser = argparse.ArgumentParser(description = "Run tracker")
+    parser.add_argument("--net", type = str, help = "Full path to onnx model of net")
+    parser.add_argument("--kernel_r1", type = str, help = "Full path to onnx model of kernel_r1")
+    parser.add_argument("--kernel_cls1", type = str, help = "Full path to onnx model of kernel_cls1")
+    parser.add_argument("--input", type = str, default = " ", help = "Full path to input. Do not use if input is camera")
+    args = parser.parse_args()
+
+    point1 = ()
+    point2 = ()
+    mark = True
+    drawing = False
+    cx, cy, w, h = 0.0, 0.0, 0, 0
+
+    def get_bb(event, x, y, flag, param):
+        nonlocal point1, point2, cx, cy, w, h, drawing, mark
+
+        if event == cv.EVENT_LBUTTONDOWN:
+            if drawing == False:
+                drawing = True
+                point1 = (x, y)
+            else:
+                drawing = False
+        elif event == cv.EVENT_MOUSEMOVE:
+            if drawing == True:
+                point2 = (x, y)
+        elif event == cv.EVENT_LBUTTONUP:
+            cx = point1[0] - (point1[0] - point2[0]) / 2
+            cy = point1[1] - (point1[1] - point2[1]) / 2
+            w = abs(point1[0] - point2[0])
+            h = abs(point1[1] - point2[1])
+            mark = False
+
+    net = cv.dnn.readNet(args.net)
+    kernel_r1 = cv.dnn.readNet(args.kernel_r1)
+    kernel_cls1 = cv.dnn.readNet(args.kernel_cls1)
+
+    if args.input == " ":
+
+        cap = cv.VideoCapture(0, cv.CAP_V4L2)
+        cap.set(cv.CAP_PROP_FRAME_HEIGHT, 480)
+        cap.set(cv.CAP_PROP_FRAME_WIDTH, 640)
+        ret, frame = cap.read()
+
+        cv.namedWindow("DaSiamRPN")
+        cv.setMouseCallback("DaSiamRPN", get_bb)
+
+        while mark == True:
+            ret, frame = cap.read()
+            if point1 and point2:
+                cv.rectangle(frame, point1, point2, (0, 255, 255), 3)
+            cv.imshow("DaSiamRPN", frame)
+            cv.waitKey(1)
+
+    target_pos, target_sz = np.array([cx, cy]), np.array([w, h])
+    Tracker = DaSiamRPNTracker(frame, target_pos, target_sz, net, kernel_r1, kernel_cls1)
+
+    while (cap.isOpened):
+        ret,frame = cap.read()
+        Tracker.track(frame)
```

Comment:
`track` should return something. It's expected to return coordinates (cx, cy, w, h). And it also should receive coordinates from the previous step.

Reply:
For now `track` change values of `self.target_pos` and `self.target_sz`. No need return values.

Reply:
It's better to return them rather than keep like a state.

Reply:
Can you please explain why it is better?